[민형] 레이싱 게임

Gobukgol opened the pull request:
1. 테스트 코드에서 막히는 중
2. 패키지 구조 아직 잘 안되어있음

pci2676 committed:
jacoco target change

 - exclude  -> include
  - domain in, view out

pci2676 committed:
Merge branch 'test'

pci2676 committed:
Merge branch 'master' of github.com:Java-Bom/racing-game

Gobukgol committed:
1. 테스트코드 작성에 어려움을 굉장히 느끼는 중

Gobukgol committed:
idea

Gobukgol committed:
1. RacingGameController 추가
2. RacingGameResult 제거, 각 이동 시도 후 결과 출력하도록 변경
3. InputValueDto 추가, 정수형 인자값 validate

chldbtjd2272 commented on src/main/java/racing/service/RacingGameService.java:
```diff
@@ -0,0 +1,31 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class RacingGameService {
```
이상한 웹레이어야, Service에 도메인 객체를 멤버변수로 들고있으면 이상해
웹계층이 아닌 도메인으로 표현해봐

chldbtjd2272 commented on src/main/java/racing/service/RacingGameService.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    private List<Car> createCars(int numberOfCars) {
+        List<Car> cars = new ArrayList<>();
```
자동차를 생성하는 로직은 ParticipateCars안쪽으로 넣는게 맞아

chldbtjd2272 commented on src/main/java/racing/RacingGameController.java:
```diff
@@ -0,0 +1,33 @@
+
+import java.util.List;
+
+public class RacingGameController {
```
웹레이어와 맞지않는 형태야

chldbtjd2272 commented on src/main/java/racing/RacingGameController.java:
```diff
@@ -0,0 +1,33 @@
+        this.racingGameService = new RacingGameService(inputValueDto.getNumberOfCars());
+    }
+
+    public void startGame() {
```
레이스를 돌리는 로직도 도메인영역에 표현되어있어야해

chldbtjd2272 commented on src/main/java/racing/domain/ParticipateCars.java:
```diff
@@ -0,0 +1,20 @@
+        this.cars = cars;
+    }
+
+    public List<Integer> tryMove(NumberGenerator numberGenerator) {
```
원시타입을 반환하는 형태보단 결과객체를 반환하는 형태가좋을듯
예를들어 자동차 주인이름이라는 요구사항이 추가되면 어떤형태가 더 기능확장하기 좋을지 생각해봐

Gobukgol committed:
요구사항 1 리팩토링
1. Service, Controller 붙어있는 클래스명 변경(웹 계층 -> 도메인)
2. 각 이동 결과를 가지고있는 RacingResult.java 추가
3. 각 RacingResult.java 를 가지고있는 GameResult.java 추가

Gobukgol committed:
요구사항 2 구현
1. 각 시도 결과를 가지고 있는 PhaseResult 추가
   - 각 PhaseResult는 자동차의 이름을 Key, 이동 결과를 Value로 하는 Map을 가지고 있음
   - PhaseResult는 현재 선두의 이름을 List의 형태로 반환하는 메소드를 가지고있음
2. PhaseResult들을 가지고있는 총 게임 결과 GameResult 추가
   - 시도 번호로 각 시도 결과를 가져올 수 있음
   - 게임 우승자는 마지막 PhaseResult의 선두의 이름을 가져옴
3. ParticipateCars 의 주석은 어떤게 더 보기좋은지 궁금해서 주석 처리하였음
